### PR TITLE
UI Code for V.1.3 Features

### PR DESCRIPTION
**Pull request by JamesLiuZX**
Popup:
- [x] Added RecipePopup.fxml
- [x] Added grid view for UI 
- [x] Add keyboard listener for Recipe Popup (Exit popup) 
- [x] Reposition/remove 'close' button in Popup (Remove since its terminal-based app)
- [x] Modal + data responsiveness 

Form:
- [x] Added RecipeForm.fxml
- [x] Added RecipeForm.java with save changes button and fields. 
- [x] Add 'ESC' key listener to close form
- [x] Add EditRecipeEvent.java
- [x] Add handleEditRecipeEvent() in MainWindow.java (parsing the changed form fields an executable to edit command). 

Card:
- [x] Add preceding titles for card fields
Screenshot 2023-03-19 at 9 15 41 AM

- [x] Add keyboard listener for Recipe Card (Delete) - Keyboard 'DELETE' or 'BACK_SPACE' or 'D'
- [x] Add keyboard listener for Recipe Popup - Keyboard letter 'P' 
- [x] Add keyboard listener for Recipe Edit Form - Keyboard letter 'F' 

RecipeListPanel:
- [x] Add borders around each card for greater visibility

Overall:
- [ ] Improve aesthetics of overall app (theme, colours, font and sizes)

To-Do:
- [ ] Bugfix handleEditRecipeEvent() 
- [ ] Refactor ingredient as per @SeeuSim 's PR 
- [ ] Display multiple fields for multiple steps (challenging for the FXML)

### CODE REVIEW

**Comment by SeeuSim** [comment]
`Optional` pipelines may have to be re-integrated into models and the unit-tests be added for them, if they are necessary.

**Review by alson001 — APPROVED** [review]
LGTM

**Review by SeeuSim — APPROVED** [review]
lgtm

File: src/main/java/seedu/recipe/ui/RecipeCard.java [hunk 1]
```diff
@@ -50,45 +60,79 @@ public class RecipeCard extends UiPart<Region> {
...
         });
+
+        // Handle keypress events
+        cardPane.setOnKeyPressed(event -> {
+            cardPane.requestFocus();
+            KeyCode input = event.getCode();
+            if (input == KeyCode.DELETE ||
+                input == KeyCode.D ||
+                input == KeyCode.BACK_SPACE) {
+                DeleteRecipeEvent deleteEvent = new DeleteRecipeEvent(displayedIndex);
```
Comment (ZHTang29) [inline src/main/java/seedu/recipe/ui/RecipeCard.java:122]: Good consideration on different keys such as backspace and delete for deleting a recipe- makes life easier especially for good typists that don't have to navigate with their mouse

File: src/main/java/seedu/recipe/ui/RecipePopup.java [hunk 2]
```diff
@@ -94,12 +95,13 @@ public void display() {
...
         window.setMinHeight(300);
-
-        Button closeButton = new Button("Close");
-        closeButton.setOnAction(event -> window.close());
-        // Create a VBox to hold the RecipeCard
-        VBox vbox = new VBox(getRoot(), closeButton);
+        VBox vbox = new VBox(getRoot());
         Scene scene = new Scene(vbox);
+        scene.setOnKeyPressed(event -> {
+            if (event.getCode() == KeyCode.ESCAPE) {
```
Comment (ZHTang29) [inline src/main/java/seedu/recipe/ui/RecipePopup.java:101]: Similarly good consideration for esc button!

**Review by ZHTang29 — COMMENTED** [review]
Overall good quality of work! Especially with the considerations with the key presses for experienced typists. The form for edits looks good to me too and it seems functional, and good consideration for having a new line for every variable field.

**Pull request merged**